docs/_future/custom_functions.md

```diff
@@ -0,0 +1,215 @@
+| `title` | A required string to nicely identify the function. |
+| `description` | An _optional_ succinct summary of what your function does. |
+| `input_parameters` | An object which describes one or more input parameters that will be available to your function. Each top-level property of this object defines the name of one input parameter which will become available to your function. 
+| `output_parameters` | An object which describes one or more output parameters that will be returned by your function. This object follows the exact same pattern as `input_parameters`: top-level properties of the object define output parameter names, with the property values further describe the type and description of each individual output parameter. |
+
+The value for properties in `input_parameters` and `output_parameters` needs to be an object with further sub-properties:
+  * `type`: The type of the input parameter. This can be a [Built-in type](https://api.slack.com/future/types) or a [Custom type](https://api.slack.com/future/types/custom) that you define.
+  * `description`: A string description of the parameter.
+
+If you want to set a property as required, list its name in its respective input or output properties `required` property.
+
+For example, if you have an input parameter named `customer_id` that you want to be required, you can do so like this:
+
+```javascript
+input_parameters: {
+  properties: {
+    customer_id: {
+      type: Schema.types.string,
+      description: "The customer's ID"
+    }
+  },
+  required: ["customer_id"]
+}
+```
+
+Functions can (and generally should) declare inputs and outputs. 
+
+Inputs are declared in the `input_parameters` property, and outputs are declared in the `output_parameters` property. Each can contain either [Built-in types](https://api.slack.com/future/types) or [Custom types](https://api.slack.com/future/types/custom) that you define.
+
+While, strictly speaking, input and output parameters are optional, they are a common and standard way to pass data between functions and nearly any function you write will expect at least one input and pass along an output.
+
+Functions are similar in philosophy to Unix system commands: they should be minimalist, modular, and reusable. Expect the output of one function to eventually become the input of another, with no other frame of reference.
+
+Once your function is defined in its own file in `manifest/function`, the next steps are to add a function listener, register the listener, and implement the function in a workflow.
+
+---
+
+### Implementing a function {#implement}
```

Comment:
```suggestion
### Implementing a Function {#implement}
```
Ooh ought these be capital? I don't really know...

Reply:
i think it makes sense for them to be capital!!